jvcl/run/JvMemoryDataset.pas

```diff
@@ -833,7 +834,11 @@ procedure TJvMemoryData.InitFieldDefsFromFields;
     begin
       FOffsets[I] := Offset;
       if FieldDefList[I].DataType in ftSupported - ftBlobTypes then
-        Inc(Offset, CalcFieldLen(FieldDefList[I].DataType, FieldDefList[I].Size) + 1);
+      begin
+        FieldLen := CalcFieldLen(FieldDefList[I].DataType, FieldDefList[I].Size);
+        if Offset + FieldLen + 1 <= high(Offset) then
+          Inc(Offset, FieldLen + 1);
+      end;
```

Comment:
What happens if the test is `False`?
Wouldn't it be an error situation that should be reported via an exception?

Reply:
Yes that would be a good idea. But which one to throw? ERangeError or EIntOverflow?